### PR TITLE
Added accordion, refactored editable text along with event propagation behaviour

### PR DESCRIPTION
**Pull request by elayabharath**

### CODE REVIEW

File: ui/lib/components/accordion/accordion.dart [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import 'dart:html';
+
+/*
+<div class="accordion">
+  <div class="accordion-item">
+    <div class="accordion-header__wrapper">
```
Comment (marianamarasoiu) [inline ui/lib/components/accordion/accordion.dart:6]: nit - what do you think about this name?
```suggestion
    <div class="accordion-item__header">
```
and similarly for the body wrapper below - `accordion-item__body`

File: ui/lib/components/accordion/accordion.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+.accordion-item:first-child {
+  border-top: 1px solid #cccc;
+}
+
+.accordion-header__wrapper {
+  padding: 6px 12px;
+  cursor: pointer;
+}
+
+.accordion-header__wrapper > div{
```
Comment (marianamarasoiu) [inline ui/lib/components/accordion/accordion.css:20]: nit:
```suggestion
.accordion-header__wrapper > div {
```

File: ui/lib/components/accordion/accordion.dart (hunk 1 above)
Comment (marianamarasoiu) [reply ui/lib/components/accordion/accordion.dart:6]: (and I think you can keep the `_headerWrapper` name below, this is just a comment for the css class)

File: ui/lib/components/accordion/accordion.dart [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        item.collapse();
+      }
+      if (expandAtStart) {
+        item.expand();
+      }
+    }
+  }
+
+  AccordionItem queryItem(String id) {
+    return _accordionItems.firstWhere((item) => item.id == id);
```
Comment (marianamarasoiu) [inline ui/lib/components/accordion/accordion.dart:103]: can you add a bit of a defense here to return `null` if there's no item with that id?

**Review by marianamarasoiu — APPROVED** [review]
LGTM, thanks @elayabharath !  just a couple of small comments for you to consider, thanks!

**Comment by elayabharath** [comment]
Thanks @marianamarasoiu Addressed the comments

**Pull request merged**